Lib/statistics.py

```diff
@@ -355,6 +355,31 @@ def _fail_neg(values, errmsg='negative value'):
             raise StatisticsError(errmsg)
         yield x
 
+def _rank(data, /) -> list[float]:
+    """Rank order a dataset. The lowest value has rank 1.
+
+    Ties are averaged so that equal values receive the same rank.
+    The operation is idempotent.
+
+    >>> data = [31, 56, 31, 25, 75, 18]
+    >>> _rank(data)
+    [3.5, 5.0, 3.5, 2.0, 6.0, 1.0]
+    >>> _rank(_)
```

Comment:
Is the special behaviour of `_` guaranteed by doctest? I suppose this is safe enough.

Reply:
Yes.